File: scenes-feed-video/src/main/java/net/polyv/android/player/scenes/feed/pager/PLVMediaPlayerFeedViewPager.java
package net.polyv.android.player.scenes.feed.pager;

import static net.polyv.android.player.sdk.foundation.graphics.DisplaysKt.isLandscape;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import net.polyv.android.player.common.utils.feed.PLVFeedViewPager;

/**
 * @author Hoshiiro
 */
public class PLVMediaPlayerFeedViewPager extends PLVFeedViewPager {

    private boolean isRequestDisallowInterceptTouchEvent = false;

    public PLVMediaPlayerFeedViewPager(Context context) {
        super(context);
    }

    public PLVMediaPlayerFeedViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public void requestDisallowInterceptTouchEvent(boolean disallowIntercept) {
        super.requestDisallowInterceptTouchEvent(disallowIntercept);
        this.isRequestDisallowInterceptTouchEvent = disallowIntercept;
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (isLandscape() || isRequestDisallowInterceptTouchEvent) {
            return false;
        }
        return super.onInterceptTouchEvent(ev);
    }

}

File: scenes-feed-video/src/main/java/net/polyv/android/player/scenes/feed/pager/PLVMediaPlayerFeedViewPager.kt
package net.polyv.android.player.scenes.feed.pager

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import net.polyv.android.player.common.utils.feed.PLVFeedViewPager
import net.polyv.android.player.sdk.foundation.graphics.isLandscape

/**
 * @author Hoshiiro
 */
class PLVMediaPlayerFeedViewPager @JvmOverloads constructor(
    context: Context?,
    attrs: AttributeSet? = null
) : PLVFeedViewPager(context, attrs) {

    private var isRequestDisallowInterceptTouchEvent = false

    override fun requestDisallowInterceptTouchEvent(disallowIntercept: Boolean) {
        super.requestDisallowInterceptTouchEvent(disallowIntercept)
        this.isRequestDisallowInterceptTouchEvent = disallowIntercept
    }

    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
        if (isLandscape() || isRequestDisallowInterceptTouchEvent) {
            return false
        }
        return super.onInterceptTouchEvent(ev)
    }

}
